File: CPP/No789.cc
/**
 * Created by Xiaozhong on 2020/12/14.
 * Copyright (c) 2020/12/14 Xiaozhong. All rights reserved.
 */
#include <vector>
#include <iostream>

using namespace std;

class Solution {
public:
    bool escapeGhosts(vector<vector<int>> &ghosts, vector<int> &target) {
        for (const vector<int> &ghost : ghosts) {
            if (distance({0, 0}, target) >= distance(ghost, target)) return false;
        }
        return true;
    }

private:
    int distance(vector<int> src, const vector<int> &dst) {
        return abs(src[0] - dst[0]) + abs(src[1] - dst[1]);
    }
};

File: Java/src/No789.java
/*
 * Copyright (c) 8/22/2021, HuXiaozhong. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 */

public class No789 {
    public boolean escapeGhosts(int[][] ghosts, int[] target) {
        int[] source = {0, 0};
        int dist = manhattanDistance(source, target);
        for (int[] ghost : ghosts) if (manhattanDistance(ghost, target) <= dist) return false;
        return true;
    }

    private int manhattanDistance(int[] source, int[] destination) {
        return Math.abs(source[0] - destination[0]) + Math.abs(source[1] - destination[1]);
    }

}
